src/components/ItemList/index.jsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState, useEffect } from 'react';
+import { updateItemDate } from '../Utils/firestore';
+import { differenceInHours } from 'date-fns';
+
+function ItemList({ itemName, docId, lastPurchasedDate }) {
+  const [isChecked, setIsChecked] = useState(false);
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [token] = useState(() => window.localStorage.getItem('tcl18-token'));
+  const toggle = () => setIsChecked(!isChecked);
+
+  const editProduct = (id) => {
+    updateItemDate(token, id)
+      .then(() => {
+        console.log('product updated');
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };
+
+  useEffect(() => {
+    if (isChecked) {
+      updateItemDate(token, docId).then(() =>
+        console.log('The date was updated'),
+      );
+    }
+  }, [isChecked, docId, token]);
+
+  useEffect(() => {
+    if (lastPurchasedDate && isDisabled === false) {
+      const hoursAtTheMoment = differenceInHours(
+        new Date(),
```

Comment:
Maybe, why you didn't use any variable to store the current date, I think you'll have many `new Date()` every time to verify the 24 hours?

Reply:
Ok, this will be a good improvement. thanks.